extensions/resource-deployment/src/test/azdataService.test.ts

```diff
@@ -0,0 +1,93 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import 'mocha';
+import * as TypeMoq from 'typemoq';
+import assert = require('assert');
+import { IPlatformService, CommandOptions } from '../services/platformService';
+import { AzdataService } from '../services/azdataService';
+import { BdcDeploymentType } from '../interfaces';
+
+suite('azdata service Tests', function (): void {
+	test('azdata service handles deployment types properly', async () => {
+		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
+		const azdataService = new AzdataService(mockPlatformService.object);
+		mockPlatformService.setup((service) => service.runCommand(TypeMoq.It.isAnyString(), TypeMoq.It.isAny())).returns((command: string, options: CommandOptions | undefined) => {
+			return new Promise<string>((resolve) => {
+				resolve('{"result":[]}');
+			});
+		});
+
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingAKS);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingARO);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingKubeAdm);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.ExistingOpenShift);
+		azdataService.getDeploymentProfiles(BdcDeploymentType.NewAKS);
+
+		try {
+			azdataService.getDeploymentProfiles(<BdcDeploymentType>'no-such-type');
+			throw new Error('Expecting an error but the error is not thrown');
+		}
+		catch {
+		}
+
+		mockPlatformService.verify((service) => service.runCommand(TypeMoq.It.isAnyString(), TypeMoq.It.isAny()), TypeMoq.Times.exactly(5));
+	});
+
+	test('azdata service returns correct deployment profiles', async () => {
+		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
+		const azdataService = new AzdataService(mockPlatformService.object);
+		mockPlatformService.setup((service => service.storagePath())).returns(() => {
+			return '';
+		});
+		mockPlatformService.setup((service => service.readTextFile(TypeMoq.It.isAnyString()))).returns((path: string) => {
+			return new Promise<string>((resolve) => {
+				resolve('{}');
+			});
+		});
+		mockPlatformService.setup((service) => service.runCommand(TypeMoq.It.isAnyString(), TypeMoq.It.isAny())).returns((command: string, options: CommandOptions | undefined) => {
+			return new Promise<string>((resolve, reject) => {
+				if (command === 'azdata bdc config list -o json') {
+					resolve('{"result":["aks-1","profile-2"]}');
+				} else if (command.startsWith('azdata bdc config init')) {
+					resolve(undefined);
+				}
+				else {
+					reject(`unexpected command: ${command}`);
+				}
+			});
+		});
+		const profiles = await azdataService.getDeploymentProfiles(BdcDeploymentType.NewAKS);
+		assert(profiles.length === 1);
+	});
+
+	test('get bdc endpoints', async () => {
+		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
+		const azdataService = new AzdataService(mockPlatformService.object);
```

Comment:
Any reason not to just pull this out into a top level `before` so you only initialize all this stuff once? It helps to keep the test functions shorter and thus easier to tell what exactly they're testing.

Reply:
I need to mock different behaviors for different tests, if i do that i will need to reset the mock for each tests and may end up more code.

Reply:
I'm talking about creating the objects. Then just have an `afterEach` that runs reset on the ones that have different logic. Definitely not going to be more code that way.

Reply:
i will use it when I have a lot of test cases in a file.